internal/receiver/smartagentreceiver/output.go

```diff
@@ -95,6 +88,69 @@ func getMetadataExporters(config Config, host component.Host, nextConsumer consu
 	return exporters
 }
 
+// getLogsConsumers walks through obtained Config.MetadataClients and returns all matching registered LogsConsumers,
+// if any.  At this time the SignalFx exporter is the only real target use case, but it's unexported and
+// as implemented all specified combination MetricsExporters and LogsConsumers will be returned.
+func getLogsConsumers(
+	config Config, host component.Host, nextConsumer *consumer.MetricsConsumer, logger *zap.Logger,
+) []*consumer.LogsConsumer {
+	var consumers []*consumer.LogsConsumer
+
+	metadataClients := getMetadataClients(config, host, nextConsumer, logger)
+	for _, client := range metadataClients {
+		if logsExporter, ok := (*client).(consumer.LogsConsumer); ok {
+			consumers = append(consumers, &logsExporter)
+		} else {
+			logger.Info("cannot send events to metadataClient", zap.Any("client", *client))
+		}
+	}
+
+	if len(consumers) == 0 {
+		logger.Debug("no SFx events are possible as no valid metadataClients have been provided and next pipeline component isn't a LogsConsumer")
```

Comment:
I may have missed this when it was discussed but why don't we have separate fields for metadata clients and log consumers in the config?

Reply:
They are almost certainly the same exporter, and when OTel has entities or another change to trigger removing the experimental metric metadata update consumer and open their functionality to other components beyond SFx exporter, we'll likely gain another configuration method via pipelines or component sets.

Reply:
Updated per our discussion